pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -167,44 +211,63 @@ def target(self):
+        """ Get an existing sequencer or create a new one given the (topic,
+            ordering_key) pair.
+        """
+        sequencer_key = (topic, ordering_key)
+        sequencer = self._sequencers.get(sequencer_key)
+        if sequencer is None:
+            if ordering_key == "":
+                # TODO: Consider implementing _delete_sequencer for unordered
+                # sequencer.
+                sequencer = unordered_sequencer.UnorderedSequencer(self, topic)
+            else:
+                sequencer = ordered_sequencer.OrderedSequencer(
+                    self, topic, ordering_key, self._delete_sequencer
+                )
+            self._sequencers[sequencer_key] = sequencer
+
+        return sequencer
 
-        This will create a new batch if ``create=True`` or if no batch
-        currently exists.
+    def resume_publish(self, topic, ordering_key):
+        """ Resume publish on an ordering key that has had unrecoverable errors.
 
         Args:
-            topic (str): A string representing the topic.
-            create (bool): Whether to create a new batch. Defaults to
-                :data:`False`. If :data:`True`, this will create a new batch
-                even if one already exists.
-            autocommit (bool): Whether to autocommit this batch. This is
-                primarily useful for debugging and testing, since it allows
-                the caller to avoid some side effects that batch creation
-                might have (e.g. spawning a worker to publish a batch).
+            topic (str): The topic to publish messages to.
+            ordering_key: A string that identifies related messages for which
+                publish order should be respected.
 
-        Returns:
-            ~.pubsub_v1._batch.Batch: The batch object.
+        Raises:
+            RuntimeError:
+                If called after publisher has been stopped by a `stop()` method
+                call.
+            ValueError:
+                If the topic/ordering key combination has not been seen before
+                by this client.
         """
-        # If there is no matching batch yet, then potentially create one
-        # and place it on the batches dictionary.
-        if not create:
-            batch = self._batches.get(topic)
-            if batch is None:
-                create = True
-
-        if create:
-            batch = self._batch_class(
-                autocommit=autocommit,
-                client=self,
-                settings=self.batch_settings,
-                topic=topic,
-            )
-            self._batches[topic] = batch
+        with self._batch_lock:
+            if self._is_stopped:
+                raise RuntimeError("Cannot resume publish on a stopped publisher.")
 
-        return batch
+            sequencer_key = (topic, ordering_key)
+            sequencer = self._sequencers.get(sequencer_key)
+            if sequencer is None:
```

Comment:
Is it necessary for this to be an error? Seems like it is reasonable just to silently let it through in this case.